### DISCUSSION TITLE
Power cycle required after running esphome?

### DISCUSSION THREAD

**Discussion by eigenphase** [emporia-vue-local/esphome#135] [Q&A]
It seems whenever I update my firmware via `esphome run vue2.yaml` the Vue does not respond any more. After I manually power cycle it works again.

Is this intended behavior?

Because it's really a bit problematic: Once the thing is in the panel, this step is not as "easy" any more. And since all sensors are configured via the yaml file this step will occur pretty frequently...

**Comment by eigenphase** [comment]
Bumping this …

Even if there is no solution I’d be curious if this is expected behavior anyway or if I am the only one experiencing this?

**Reply by canuckray** [reply]
I can't say that I have experienced this on my Gen 2 Vue running esphome.

**Reply by flaviut** [reply]
Not an issue for me either. Any errors or warnings should be visible in the serial console if you want to debug this.

You'd have to be very quick starting the serial console instantly after finishing flashing.

**Reply by eigenphase** [reply]
Ah bummer :-( But let me clarify: I am not talking about the serial flash (that I only did initially). Now when I do `esphome run vue2.yaml` the OTA update succeds but then the device seems to hang and does not produce any output:

```
[...]
======================================================= [SUCCESS] Took 279.37 seconds =======================================================
INFO Successfully compiled program.
INFO Connecting to 192.168.222.6
INFO Uploading .esphome/build/emporiavue2/.pioenvs/emporiavue2/firmware.bin (1018384 bytes)
Uploading: [============================================================] 100% Done...

INFO Waiting for result...
INFO OTA successful
INFO Successfully uploaded program.
INFO Starting log output from 192.168.222.6 using esphome API
WARNING Can't connect to ESPHome API for 192.168.222.6: Timeout while connecting to ('192.168.222.6', 6053)
INFO Trying to reconnect to 192.168.222.6 in the background
```

Then if I go the the panel, manually unplug the cable and plug it again, it starts again (with the newly flashed firmware).


Is there a way I can debug this further?

**Reply by flaviut** [reply]
Oh, if you can reproduce this with OTA, that makes it easier.

1. Remove vue from the panel
2. Power the vue from the debug header & connect to it using a serial console (probably 115200 baud?)
3. apply an OTA update
4. observe the things printed to serial and hopefully there's a useful error in there

The remote serial console doesn't work for debugging this, it's gotta be the physical serial console.

**Reply by biker2000on** [reply]
> Ah bummer :-( But let me clarify: I am not talking about the serial flash (that I only did initially). Now when I do `esphome run vue2.yaml` the OTA update succeds but then the device seems to hang and does not produce any output:
> 
> ```
> [...]
> ======================================================= [SUCCESS] Took 279.37 seconds =======================================================
> INFO Successfully compiled program.
> INFO Connecting to 192.168.222.6
> INFO Uploading .esphome/build/emporiavue2/.pioenvs/emporiavue2/firmware.bin (1018384 bytes)
> Uploading: [============================================================] 100% Done...
> 
> INFO Waiting for result...
> INFO OTA successful
> INFO Successfully uploaded program.
> INFO Starting log output from 192.168.222.6 using esphome API
> WARNING Can't connect to ESPHome API for 192.168.222.6: Timeout while connecting to ('192.168.222.6', 6053)
> INFO Trying to reconnect to 192.168.222.6 in the background
> ```
> 
> Then if I go the the panel, manually unplug the cable and plug it again, it starts again (with the newly flashed firmware).
> 
> Is there a way I can debug this further?

This just happened to me as well. Same exact phenomenon. Used esphome dashboard to apply an OTA update and it stopped after saying `INFO OTA successful`.

**Reply by eigenphase** [reply]
> This just happened to me as well. Same exact phenomenon.

Sorry to say but I'm happy that I'm not the only one experiences this. Is this consistent for you or just happens some times?
For me, it happens nearly all the time. 1/10 times maybe it continues afterwards.

> Oh, if you can reproduce this with OTA, that makes it easier.

@flaviut  I'm not sure if so much easier ;-) When I first flashed I used the serial adapter to power it via 3.3V (5V did actually NOT work) but that supply is not enough for the wifi module.

> Power the vue from the debug header & connect to it using a serial console (probably 115200 baud?)

Is this different from the [serial pin header](https://camo.githubusercontent.com/85adc50093dbc9d7841b2fa359990cb04558e8ee6353cf0826ccc9a68b392d87/68747470733a2f2f692e696d6775722e636f6d2f4e65745673516f2e6a706567)  ?

If no, I doubt I can do an OTA flash without powering the whole unit.
If yes, do you have a pointer how to connect it?

**Reply by flaviut** [reply]
> Is this different from the [serial pin header](https://camo.githubusercontent.com/85adc50093dbc9d7841b2fa359990cb04558e8ee6353cf0826ccc9a68b392d87/68747470733a2f2f692e696d6775722e636f6d2f4e65745673516f2e6a706567) ?

That's exactly the right header.

Anyway, no idea why the brownouts are happening. Just hooked it up to my test equipment, and other than 850mA being drawn for 8.5ms to charge up the capacitors, all power consumption is under the 500mA that USB allows. I've experienced brownouts when connecting over a serial dongle too, by the way.

Not sure what stuff you have, but what if you power your Vue via 5V with a USB wall brick, and just use the computer's USB for serial? You would need to make sure the two ground connections (wall brick & computer) are connected. Not sure why 5V didn't work when you tried it.

---

Also, I strongly do not recommend this, and in europe (240V) I would never do this myself despite my extensive electronics experience, but it is technically possible to power the Vue off mains electricity while having it connected to a computer. The mains side is isolated on the Vue. Note that this is not the case for. for example. wall switches and that trying it on them will, at best, destroy your computer.

**Comment by philsson** [comment]
I came here looking for an answer to this. I have over 30 esphome devices at home and they are 2 of them that regularly require a power cycle after updating, all of which are connected to mains. Affected devices are one sonoff wall switch and a shelly 1L. I have multiple of these devices and none the others experience this problem.
